src/spyglass/behavior/moseq.py

```diff
@@ -0,0 +1,360 @@
+import hashlib
+import os
+from pathlib import Path
+
+import datajoint as dj
+import keypoint_moseq as kpms
+
+from spyglass.common import AnalysisNwbfile
+from spyglass.position.position_merge import PoseOutput
+from spyglass.utils import SpyglassMixin
+
+from .core import PoseGroup, format_dataset_for_moseq, results_to_df
+
+schema = dj.schema("moseq_v1")
+
+
+@schema
+class MoseqModelParams(SpyglassMixin, dj.Manual):
+    definition = """
+    model_params_name: varchar(80)
+    ---
+    model_params: longblob
+    """
+
+    def make_training_extension_params(
+        self, model_key, num_epochs, new_name=None, skip_duplicates=False
+    ):
+        """Method to create a new set of model parameters for extending training
+
+        Parameters
+        ----------
+        model_key : dict
+            key to a single MoseqModelSelection table entry
+        num_epochs : int
+            number of epochs to extend training by
+        new_name : str, optional
+            name for the new model parameters, by default None
+        skip_duplicates : bool, optional
+            whether to skip duplicates, by default False
+        Returns
+        -------
+        dict
+            key to a single MoseqModelParams table entry
+        """
+        model_key = (MoseqModel & model_key).fetch1("KEY")
+        model_params = (self & model_key).fetch1("model_params")
+        model_params["num_epochs"] = num_epochs
+        model_params["initial_model"] = model_key
+        # increment param name
+        if new_name is None:
+            # increment the extension number
+            if model_key["model_params_name"][:-1].endswith("_extension"):
```

Comment:
if this moseq's convention? using the `extension` suffix? I'd avoid mixing data types like this if possible

Reply:
This is part of a system I made to allow for extended training of an existing model (defined by `model key`)

There isn't anything particular to the term extension, but it allows me to have this helper function for users to build an extended traning